Config.cmake

```diff
@@ -68,11 +68,21 @@ set(ODBFLAGS
   --include-with-brackets
   --default-pointer "std::shared_ptr")
 
+# Set CXX flags
 set(CMAKE_CXX_FLAGS "-W -Wall -Wextra -pedantic\
   -std=c++14 \
   -DDATABASE_${DATABASE_U} \
   -DBOOST_LOG_DYN_LINK")
 
+# Gold is the primary linker 
+if(NOT DEFINED CODECOMPASS_LINKER)
+    set(CODECOMPASS_LINKER "/usr/bin/gold")
```

Comment:
I think we shouldn't hard-code this path. We should introduce some CMake magic here to find `gold` binary. Probably there are specific CMake commands for this. Moreover in case this linker is not installed for some reason on a platform we should leave `$CMAKE_LINKER` on its default value.

Reply:
Sadly I didn't find any of that kind of magic. Cmake only has compiler auto detection, and then the compiler driver picks the linker if we don't specify it. There is one way where, we make an extrenal tool chain (like with cross compiling) and implement the linker selection in that. Then we pass this tool chain to the compiler driver. I don't believe this should be worth it.

Reply:
Here's the relevant magic from LLVM tree: https://reviews.llvm.org/D28738

Reply:
How is this is any magic? As far as I can see, it simply uses the `lld` as the linker's binary path, which has to be found on the `PATH` I guess. There is no work done on locating this binary.